### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by larsgrefer**
I simply ran IntelliJ's code inspection and applied the quick fixes.

### CODE REVIEW

**Comment by andrebrait** [comment]
So you're the guys who created IntelliJ IDEA's inspections! Gotcha!
:)

Jokes aside, nice work there.

**Comment by lukehutch** [comment]
Me, this is a great contribution. I think some of these are JDK 8+ only (eg. String constants in switch cases?), and right now JDK 7 is supported, but I think it's possible to compile at a higher source level than the target level? I'll have to look into this a bit before merging (or maybe I can just merge and then fix whatever is broken...)

What tool did you use to find these? Just IntelliJ? Currently I'm using Eclipse + LGTM for static analysis, but maybe I should be using IntelliJ too. I wonder if it's possible to run IntelliJ's static code analysis from Eclipse as a plugin :o)

**Comment by lukehutch** [comment]
Also, how did you turn each fix into a separate commit? Is this an IntelliJ feature too?

**Comment by larsgrefer** [comment]
These are just some of the default inspections of IntelliJ Ultimate 2018.2.5 (IntelliJ community should work the same but you might also qualify for https://www.jetbrains.com/buy/opensource/ )

this is everything java <=7 (IntelliJ skipped the 8+ checks because the compile level is 7)

The separate commits were my manual work.

File: src/main/java/io/github/classgraph/AnnotationParameterValueList.java [hunk 1]
```diff
@@ -154,8 +154,7 @@ public AnnotationParameterValue set(final int index, final AnnotationParameterVa
      * to check the type of each method of the annotation class to determine if it is a primitive array type).
      */
     void convertWrapperArraysToPrimitiveArrays(final ClassInfo annotationClassInfo) {
-        for (int i = 0; i < size(); i++) {
-            final AnnotationParameterValue annotationParamValue = get(i);
```
[the remark this answers is not in the document]
Comment (larsgrefer) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: _Works on my machine_™

File: src/main/java/io/github/classgraph/AnnotationParameterValueList.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public AnnotationParameterValue set(final int index, final AnnotationParameterVa
      */
     public List<String> getNames() {
         if (this.isEmpty()) {
-            return Collections.<String> emptyList();
```
[the remark this answers is not in the document]
Comment (larsgrefer) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: _Works on my machine_™

[the remark this answers is not in the document]
Comment (larsgrefer) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: And works on travis

File: src/main/java/io/github/classgraph/AnnotationParameterValueList.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (larsgrefer) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: The iterator of `ArrayList` effectively uses an index based approach too

File: src/main/java/io/github/classgraph/AnnotationParameterValueList.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukehutch) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: OK, it must have been some other type parameters that I was running into this with then. The Eclipse compiler happily removed them, but the Maven compiler errored.

File: src/main/java/io/github/classgraph/AnnotationParameterValueList.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukehutch) [reply src/main/java/io/github/classgraph/AnnotationParameterValueList.java]: Right, but it allocates one intermediate object for the iterator. The Android recommendations say to never use foreach for this reason (to save on battery life). Not a big deal though.

**Comment by lukehutch** [comment]
Merging, thanks again for these cleanups!

**Comment by larsgrefer** [comment]
You're welcome

**Comment by lukehutch** [comment]
PS it looks like you were still pushing more cleanups... feel free to open another PR if you find more.

**Comment by andrebrait** [comment]
For collections, a foreach loop will almost always be faster than using indexes and ```get(int i)```. For arrays, using an index is faster.

It's weird that Android recommends using indexes on collections, given these are only suitable for specific implementations of lists which are backed by an array (like, say, ArrayList).

**Pull request merged**